src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/WorkspaceProjectStateChangeDetector.cs

```diff
@@ -116,13 +118,21 @@ internal virtual void Workspace_WorkspaceChanged(object sender, WorkspaceChangeE
 
                         if (document.FilePath == null)
                         {
-                            break;
+                            return;
                         }
 
                         // Using EndsWith because Path.GetExtension will ignore everything before .cs
                         // Using Ordinal because the SDK generates these filenames.
                         // Stll have .cshtml.g.cs and .razor.g.cs for Razor.VSCode scenarios.
                         if (document.FilePath.EndsWith(".cshtml.g.cs", StringComparison.Ordinal) || document.FilePath.EndsWith(".razor.g.cs", StringComparison.Ordinal) || document.FilePath.EndsWith(".razor", StringComparison.Ordinal))
+                        {
+                            EnqueueUpdate(e.ProjectId);
+                            return;
+                        }
+
+                        // We now know we're not operating directly on a Razor file. However, it's possible the user is operating on a partial class that is associated with a Razor file.
+
+                        if (IsPartialComponentClass(document))
```

Comment:
Hmm as far as I can tell this gets called for every C# file on every keystroke. @NTaylorMullen am I holding it wrong? :)